lib/bridge/MatrixHandler.js

```diff
@@ -770,6 +769,68 @@ MatrixHandler.prototype._onMessage = Promise.coroutine(function*(req, event) {
     yield Promise.all(promises);
 });
 
+MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
+    function*(req, ircRoom, ircUser, ircAction, event) {
+    let self = this;
+
+    if (event.content.msgtype !== "m.text" ||
+        !(ircUser.unsafeClient && ircUser.unsafeClient.wouldSend)) {
+        yield self.ircBridge.sendIrcAction(ircRoom, ircUser, ircAction);
+    }
+    else {
```

Comment:
As a rule of thumb, try to return early so you can outdent e.g this:
```javascript
function foo() {
  if (some condition) {
    // if block
    return;
  }
  // else block goes here
}
```

Rather than what you have here:
```javascript
function foo() {
  if (some condition) {
    // if block
  }
  else {
    // else block
  }
}
```

Two reasons for this:
 - Less indentation (which is always good)
 - Less chance of you accidentally executing code on the `if` block you weren't expecting to e.g:

    ```javascript
    function foo() {
      if (some condition) {
        // if block
      }
      else {
        // else block
      }
      // extra code you didn't mean to execute on the if block
    }
```

You will also find this follows your mental model a lot better as the "success path" of the function tends down the page rather than being sprawled with splits from conditional statements.